src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.IDictionary.cs

```diff
@@ -62,12 +62,16 @@ public bool TryAdd(string propertyName, JsonNode? value, out int index)
                 ThrowHelper.ThrowArgumentNullException(nameof(propertyName));
             }
 
+#if NET10_0_OR_GREATER
```

Comment:
We only need this for .NET 9. Earlier targets polyfill the current OrderedDictionary implementation.
```suggestion
#if NET9_0
```

Reply:
I assume it becomes this then?
```c#
#if NET9_0
            var success = Dictionary.TryAdd(propertyName, value);
            index = Dictionary.IndexOf(propertyName);
#else
            var success = Dictionary.TryAdd(propertyName, value, out index);
#endif
```

Reply:
Right.